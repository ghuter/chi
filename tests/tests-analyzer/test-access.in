Point :: struct {
	x: i32;
	y: i32;
}

f :: fun () {
	p := Point {
		x: 0,
		y: 0,
	};

	p.x = 10;
	x :: p.x;
}
