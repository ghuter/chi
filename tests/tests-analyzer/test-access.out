SSTRUCT(Point, x : i32, y : i32)
SFUN(f, {SSEQ(SDECL(p, ESTRUCT(Point { <x : ECSTI(0)> <y : ECSTI(0)> })) : Point, SSEQ(SEXPRASSIGN(EACCESS(EMEM(p) : Point.x) : i32, ECSTI(10)), const SDECL(x, EACCESS(EMEM(p) : Point.x) : i32) : i32))}) : void
